xalia/Win32/HwndProvider.cs

```diff
@@ -431,11 +448,55 @@ public bool WatchProperty(UiDomElement element, GudlExpression expression)
                         if (!WindowRectKnown)
                             RefreshWindowRect();
                         return true;
+                    case "win32_window_text":
+                        if (!_fetchingWindowText)
+                        {
+                            _fetchingWindowText = true;
+                            Utils.RunTask(FetchWindowText());
+                        }
+                        break;
                 }
             }
             return false;
         }
 
+        private async Task FetchWindowText()
+        {
+            string result;
+            try
+            {
+                result = await Connection.CommandThread.OnBackgroundThread(() =>
+                {
+                    int buffer_size = 256;
+                    IntPtr buffer = Marshal.AllocCoTaskMem(buffer_size * 2);
+                    try
+                    {
+                        // For some reason SendMessageCallback refuses to send this
+                        int buffer_length = unchecked((int)(long)SendMessageW(Hwnd, WM_GETTEXT, (IntPtr)buffer_size, buffer));
+                        if (buffer_length >= 0 && buffer_length <= buffer_size)
+                            return Marshal.PtrToStringUni(buffer, buffer_length);
+                    }
+                    finally
+                    {
+                        Marshal.FreeCoTaskMem(buffer);
+                    }
+                    return null;
+                }, Tid+1);

```

Comment:
Any reason not to send `WM_GETTEXTLENGTH` first here, allocate a buffer of the size returned, and then send `WM_GETTEXT`?

Reply:
Mostly because I don't want to allocate a large buffer in the case of, say, an editbox, unless we're really sure we need it.

Reply:
In theory gudl can prevent that, but I don't trust myself to not have a stray `name` in there.

Reply:
That makes sense. I guess the names you're realistically checking for are unlikely to be more than 256 characters.